packages/apputils/src/dialog.tsx

```diff
@@ -392,7 +394,9 @@ export class Dialog<T> extends Widget {
     const target = event.target as HTMLElement;
     if (!this.node.contains(target as HTMLElement)) {
       event.stopPropagation();
-      this._buttonNodes[this._defaultButton].focus();
+      if (this._buttonNodes && this._buttonNodes[this._defaultButton]) {
+        this._buttonNodes[this._defaultButton].focus();
+      }
```

Comment:
`this._buttonNodes` is always defined although it may be an empty array in this case. So we could take advantage of the nullish operator:

```suggestion
      this._buttonNodes[this._defaultButton]?.focus();
```